test/Generators/Microsoft.Gen.MetadataExtractor/Unit/GeneratorTests.cs

```diff
@@ -80,19 +80,18 @@ public async Task TestAll(bool useExplicitReportPath)
             var goldenFileName = Path.ChangeExtension(stem, ".json");
             var goldenReportPath = Path.Combine("GoldenReports", goldenFileName);
 
-            var generatedReportPath = Path.Combine(Directory.GetCurrentDirectory(), ReportFilename);
-
             if (File.Exists(goldenReportPath))
             {
-                var d = await RunGenerator(await File.ReadAllTextAsync(inputFile), options);
+                var tmp = Path.GetTempFileName();
+                var d = await RunGenerator(await File.ReadAllTextAsync(inputFile), tmp, options: options);
```

Comment:
[style] No vars unless the type is obvious.
Please follow the [style guide](https://github.com/dotnet/extensions/blob/main/CONTRIBUTING.md#dos-and-donts)